### ISSUE TITLE
Cannot build MAME in Debian Linux 11

### ISSUE THREAD

**Issue by MiltosKoutsokeras** [mamedev/mame#10822]
### MAME version

0.251

### System information

Debian Linux 11, amd64, 32GB system RAM, Nvidia Geforce GTX 1050

### Incorrect behaviour

Cannot build the emulator or any program according to [official instructions](https://docs.mamedev.org/initialsetup/compilingmame.html#debian-and-ubuntu-including-raspberry-pi-and-odroid-devices). 

The build error is:

```
.... (lot of output here)
Compressing src/mame/layout/zodiac.lay...
Compressing src/mame/layout/zoo.lay...
None of the specified source files contain system drivers
/source_code/scripts/genie.lua:1461: Error creating projects from specified source files
stack traceback:
[C]: in function 'error'
/source_code/scripts/genie.lua:1461: in main chunk
[C]: in upvalue 'builtin_dofile'
[string "premake = { }..."]:109: in function 'dofile'
[string "_WORKING_DIR        = os.getcwd()..."]:46: in function '_premake_main'
make: *** [makefile:1281: build/projects/sdl/mame/gmake-linux/Makefile] Error 1
```

### Expected behaviour

After installing apt package dependencies, make should complete the build.

### Steps to reproduce

1. Checkout MAME code: `git clone https://github.com/mamedev/mame.git`
2. Change to version tag: `git checkout mame0251`
3. Build with make: `make REGENIE=1 TOOLS=1`

### Additional details

This is not an issue that affects only the latest 0.251 version. I have been seeing it since at least 0.249. Omitting `REGENIE=1 TOOLS=1`  does not change the behavior.

**Comment by cuavas** [comment]
What’s the contents of your environment?  Do you have `SOURCES`, `SOURCEFILTER`, `TARGET` or `SUBTARGET` environment variables set?  This should not happen for a full build.

**Comment by MiltosKoutsokeras** [comment]
After failing to build on my host, I tried to build in a fresh Docker image of Debian 11. Same behavior. By the way I have been using this same setup to build my binaries for at least 2 years and the problem was noticed in 0.249 and forward. I started debugging and it seems that Python program `makedep.py` does not complete successfully which causes the LUA script to fail. I have also tried to invoke `makedep.py` outside of the MAME build system and again fails.

Command:

```Bash
python3 ./sources/scripts/build/makedep.py -r ./sources/ sourcesproject -t mame -l ./sources/src/mame/mame.lst  ./sources/
```
